wordpress-mobile/WordPress-iOS
# Re-implement deleting media in `MediaRepository`

What says in the title.

> **Note**
> I set the milestone to 23.6, because I don't want to rush the code review. There is no pressure to push this change to production.

## Test Instructions

You can delete media in the Media screen under My Sites tab. Verify both deleting one images and deleting many images on one go works.

Tap an image on the Media screen. There is a delete button in the detail screen too. Verify that the delete button also works.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the "Test Instructions" section.

3. What automated tests I added (or what prevented me from doing so)
Unit tests.

PR submission checklist:


UI Changes testing checklist: N/A
